### PR TITLE
Feature: added security_policy_id to resource definition

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/380

Tests passes: https://github.com/vmanilo/terraform-provider-twingate/actions/runs/6900109331/job/18772776593?pr=37

## Changes
- added optional `security_policy_id` attribute to Resource definition

### CODE REVIEW

File: docs/resources/resource.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ resource "twingate_resource" "resource" {
 - `is_browser_shortcut_enabled` (Boolean) Controls whether an "Open in Browser" shortcut will be shown for this Resource in the Twingate Client.
 - `is_visible` (Boolean) Controls whether this Resource will be visible in the main Resource list in the Twingate Client.
 - `protocols` (Block List, Max: 1) Restrict access to certain protocols and ports. By default or when this argument is not defined, there is no restriction, and all protocols and ports are allowed. (see [below for nested schema](#nestedblock--protocols))
+- `security_policy_id` (String) The ID of a `twingate_security_policy` to set as this Resource's Security Policy.
```
Comment (twingate-blee) [inline docs/resources/resource.md:73]: Can we also add an example on how to use this resource under Example Usage?

For example:
data "twingate_security_policy" "test_policy" {
  name = "test policy"
}

resource "twingate_resource" "resource" {
...
  security_policy_id = data.twingate_security_policy.test_policy.id

**Comment by twingate-blee** [comment]
My resource was set to "Default Policy". I then used security_policy_id to set it to a different security policy lets call it "Test Policy". I then removed security_policy_id and ran Terraform which set the the value to null

`security_policy_id          = "U2VjdXJpdHlQb2xpY3k6Njk4MjA=" -> null`

I was thinking that my resources would go back to "Default Policy" but it stayed on "Test Policy"

For me to get back to default state I had to security_policy_id to the "Default Policy" then run terraform then I was able to delete security_policy_id.

We might want to explain that removing security_policy_id will still use the current policy and not go back to the previous policy.

File: twingate/internal/test/acctests/resource/resource_test.go [hunk 2]
```diff
@@ -2542,3 +2542,79 @@ func createResourceWithBrowserOption(name, networkName, resourceName, address st
 	}
 	`, name, networkName, resourceName, address, browserOption)
 }
+
+func TestAccTwingateResourceUpdateSecurityPolicy(t *testing.T) {
+	resourceName := test.RandomResourceName()
+	theResource := acctests.TerraformResource(resourceName)
+	remoteNetworkName := test.RandomName()
+
+	policies, err := acctests.ListSecurityPolicies()
```
Comment (romankor) [inline twingate/internal/test/acctests/resource/resource_test.go:2551]: I understand that correctly we neither support creating or reading ( data source ) any security policies through the provider ? 
We assume the user created these policies in the admin and copied the ID to the TF code ? 

@alexb-twingate any plans adding a datasource to Security Policies ?

File: twingate/internal/test/acctests/resource/resource_test.go [hunk 3]
```diff
@@ -2542,3 +2542,79 @@ func createResourceWithBrowserOption(name, networkName, resourceName, address st
...
+			{
+				Config: createResourceWithSecurityPolicy(remoteNetworkName, resourceName, ""),
+				// no changes
+				PlanOnly: true,
+			},
+		},
+	})
+}
+
+func createResourceWithSecurityPolicy(remoteNetwork, resource, policyID string) string {
```
Comment (romankor) [inline twingate/internal/test/acctests/resource/resource_test.go:2595]: Lets add a test where you actually change the security policy on a resource and see it being updated

**Review by romankor — CHANGES_REQUESTED** [review]
see comments

File: twingate/internal/test/acctests/resource/resource_test.go [hunk 4]
```diff
@@ -2542,3 +2542,79 @@ func createResourceWithBrowserOption(name, networkName, resourceName, address st
...
+					sdk.TestCheckResourceAttr(theResource, attr.SecurityPolicyID, policies[0].ID),
+				),
+			},
+			{
+				Config: createResourceWithSecurityPolicy(remoteNetworkName, resourceName, policies[1].ID),
+				Check: acctests.ComposeTestCheckFunc(
+					acctests.CheckTwingateResourceExists(theResource),
+					sdk.TestCheckResourceAttr(theResource, attr.SecurityPolicyID, policies[1].ID),
+				),
+			},
```
Comment (vmanilo) [inline twingate/internal/test/acctests/resource/resource_test.go:2578]: @romankor that's the test to check that security policy was changed on the resource
first we set policy[0], and checked it
next we updated resource with setting policy[1], and checked it

or did you mean something else?

File: docs/resources/resource.md (hunk 1 above)
Comment (vmanilo) [reply docs/resources/resource.md:79]: added example with datasource

**Comment by twingate-blee** [comment]
> My resource was set to "Default Policy". I then used security_policy_id to set it to a different security policy lets call it "Test Policy". I then removed security_policy_id and ran Terraform which set the the value to null
> 
> `security_policy_id = "U2VjdXJpdHlQb2xpY3k6Njk4MjA=" -> null`
> 
> I was thinking that my resources would go back to "Default Policy" but it stayed on "Test Policy"
> 
> For me to get back to default state I had to security_policy_id to the "Default Policy" then run terraform then I was able to delete security_policy_id.
> 
> We might want to explain that removing security_policy_id will still use the current policy and not go back to the previous policy.

I retested after your update and verified that if you remove security_policy_id it defaults to "Default Policy". I also tested setting the security policy to another policy from the UI. Ran terraform without security_policy_id and verified that it didn't try to change it to "Default Policy"

**Comment by vmanilo** [comment]
I see, working on it

**Comment by vmanilo** [comment]
hey @twingate-blee I've pushed a commit to address your comment
please give it a try

File: docs/resources/resource.md [hunk 5]
```diff
@@ -70,6 +76,7 @@ resource "twingate_resource" "resource" {
...
```
[the remark this answers is not in the document]
Comment (vmanilo) [reply docs/resources/resource.md]: updated docs

**Review by twingate-blee — APPROVED** [review]

**Review by romankor — APPROVED** [review]
GTM

**Pull request merged**